acme/provider.go

```diff
@@ -0,0 +1,7 @@
+package acme
+
+// ChallengeProvider makes the result of a challenge available to be solved
+type ChallengeProvider interface {
+	Present(domain, token, keyAuth string) error
+	CleanUp(domain string) error
```

Comment:
@xenolf a question for you is how you feel about the `CleanUp` signature here

Having only domain seems intuitively sufficient, but it turned out to be slightly awkward for the DNS provider as they need to know exact records to cleanup, and that meant they had to cache some data locally. If you would rather i switch this signature to be `CleanUp(domain, token, keyAuth string) errror` it would clean up that case.